test/LEGO.AsyncAPI.Tests/Writers/SpecialCharacterStringExtensionsTests.cs

```diff
@@ -0,0 +1,121 @@
+﻿// Copyright (c) The LEGO Group. All rights reserved.
+
+namespace LEGO.AsyncAPI.Tests.Writers
+{
+    using LEGO.AsyncAPI.Writers;
+    using NUnit.Framework;
+    using System;
+
+    internal class SpecialCharacterStringExtensionsTests
+    {
+        [Test]
+        public void GetYamlCompatibleString_NullValue_ReturnsNull()
+            => this.Compose(null, "null");
+
+        [Test]
+        public void GetYamlCompatibleString_EmptyValue_ReturnsNull()
+            => this.Compose(string.Empty, "''");
+
+        [Test]
+        public void GetYamlCompatibleString_NullWordString_ReturnsWrappedValue()
+            => this.Compose("null", "'null'");
+
+        [Test]
+        public void GetYamlCompatibleString_TildaWordString_ReturnsWrappedValue()
+              => this.Compose("~", "'~'");
+
+        // [Test]
+        // public void GetYamlCompatibleString_IntegerWithTwoPeriods_RendersPlainStyle()
+        //    => this.Compose("1.2.3", "1.2.3"); 
+        // TODO: This actually wraps in quotes which is wrong but this is already existing behaviour
```

Comment:
This actually wraps it in a quote, which is incorrect. This is because `1.2.3` is a valid datetime

Reply:
remember that thing about dates and weird formats we talked about? 😅